### ISSUE TITLE
Warning: Transaction flow is not supported.

### ISSUE THREAD

**Issue by ChiggyDhawan** [dotnet/wcf#2693]
Hi,

I am getting the below warnings when trying to add an external WCF service as a connected service to my .net core project:
Warning: Transaction flow is not supported.
Warning: Endpoint 'DefaultEndPoint1' at address 'http://hostname.co.uk/ServiceNamespace/Service.svc/Transactional' contains one or more bindings not compatible with .Net Core apps, skipping...

When I use this service by adding a web service reference to my .net 4.6.2 project, it works absolutely fine. Endpoints in my plain .net project look like:
<system.serviceModel>
    <bindings>
      <basicHttpBinding>
        <binding name="basicHttpBinding_Enquiry" transferMode="StreamedResponse" />
        <binding name="basicHttpBinding_Update" maxBufferSize="6553600" maxBufferPoolSize="524288" maxReceivedMessageSize="6553600" transferMode="StreamedRequest">
          <readerQuotas maxDepth="32" maxStringContentLength="8192000" maxArrayLength="16384" maxBytesPerRead="4096" maxNameTableCharCount="16384" />
        </binding>
        <binding name="DefaultEndPoint" closeTimeout="00:01:00" openTimeout="00:01:00" receiveTimeout="00:10:00" sendTimeout="00:01:00" allowCookies="false" bypassProxyOnLocal="false" hostNameComparisonMode="StrongWildcard" maxBufferSize="6553600" maxBufferPoolSize="524288" maxReceivedMessageSize="6553600" messageEncoding="Text" textEncoding="utf-8" transferMode="Buffered" useDefaultWebProxy="true">
          <readerQuotas maxDepth="32" maxStringContentLength="8192" maxArrayLength="16384" maxBytesPerRead="4096" maxNameTableCharCount="16384" />
        </binding>
      </basicHttpBinding>
    </bindings>
    <client>
      <endpoint address="http://hostname.co.uk/ServiceNamespace/Service.svc"
        binding="basicHttpBinding" bindingConfiguration="DefaultEndPoint"
        contract="ServiceContractName" name="DefaultEndPoint" />
    </client>
  </system.serviceModel>

Is there something I am doing wrong?
I am getting same warnings in .net Standard while consuming this service.

**Comment by mlacouture** [comment]
Hi @ChiggyDhawan 
Thanks for using the WCF Web Service Reference tool.  As the warning message implies, the [Transactions API](https://apisof.net/catalog/System.ServiceModel.TransactionFlowAttribute) is not yet available in .NET Core, unfortunately we don't have yet a road map for when it will be available.

**Comment by ShahroozAnsari** [comment]
@mlacouture  it is very important option we need in .net core client side .I need this in my project

**Comment by mconnew** [comment]
@ShahroozAnsari, WCF can't implement transaction flow because the api's in System.Transaction don't support distributed transactions, only transactions to SQL server. Before we can support transaction flow, we need proper support of System.Transaction which lives in the corefx repo. I recommend opening an issue there asking for support and if/when they provide it, open a new issue here and we'll add support. It's only about a week's work to add the implementation into WCF once there's a full transactions implementation in .Net core.
